.github/workflows/Vulnerability-Scan.yaml

```diff
@@ -34,240 +24,206 @@ on:
+    with:
+      oci-archive-name: ${{ needs.configure-scan.outputs.oci-image }}
+      test-vulnerabilities: true  # just incase we set this to false by default in the future
+      test-oci-compliance: false
+      test-efficiency: false
+      test_malware: false
 
-      - name: Process report
-        if: ${{ !cancelled() }}
-        id: check-report
-        run: |
-          report="${{ steps.vulnerability-report.outputs.name }}"
-          echo "notify=false" >> "$GITHUB_OUTPUT"
-          set -x
-          vulnerabilities="$(jq -r -c '[
-                  try(.scanner.result.Results[])
-                  | .Target as $target
-                  | .Vulnerabilities
-                  | select(. != null)
-                  | .[]
-                  | {Target: $target, LastModifiedDate: .LastModifiedDate, VulnerabilityID: .VulnerabilityID,
-                     PkgName: .PkgName, Severity: .Severity}
-                ]' < $report)"
-          echo "vulnerabilities=$vulnerabilities" >> "$GITHUB_OUTPUT"
-          last_modified_dates="$(echo "$vulnerabilities" | jq -r '.[] | select(.LastModifiedDate != null) | .LastModifiedDate')"
-          cat "$GITHUB_OUTPUT"
-          # We want to notify only if the CVEs have been updated since the last
-          # time this scan ran
-          for cve_updated in $last_modified_dates
-          do
-            if [[ "$cve_updated" > "${{ inputs.date-last-scan }}" ]]
-            then
-              echo "notify=true" >> "$GITHUB_OUTPUT"
-              break
-            fi
-          done
+  parse-results:
+    name: "parse-results ${{ inputs.oci-image-name != '' && format('| {0}', inputs.oci-image-name) || ' '}}"
+    runs-on: ubuntu-22.04
+    needs: [configure-scan, test-vulnerabilities]
+    outputs:
+      notify: ${{ steps.check-report.outputs.notify }}
+      vulnerabilities: ${{ steps.check-report.outputs.vulnerabilities }}
+    steps:
+    - uses: actions/checkout@v4
 
-      - uses: actions/cache/save@v4
-        if: ${{ !cancelled() }}
-        with:
-          path: ${{ steps.vulnerability-report.outputs.name }}
-          key: ${{ github.run_id }}-${{ steps.vulnerability-report.outputs.name }}
+    - name: Validate access to triggered image
+      uses: ./.github/actions/validate-actor
+      if: ${{ github.repository == 'canonical/oci-factory' && !github.event.pull_request.head.repo.fork }}
+      with:
+        admin-only: true
+        image-path: ${{ inputs.oci-image-path }}
+        github-token: ${{ secrets.ROCKSBOT_TOKEN }}
 
-      - uses: actions/upload-artifact@v4
-        if: ${{ !cancelled() }}
-        with:
-          name: ${{ steps.vulnerability-report.outputs.name }}
-          path: ${{ steps.vulnerability-report.outputs.name }}
+    - name: Download Vulnerability Report
+      uses: actions/download-artifact@v4
+      with:
+        name: '${{ needs.configure-scan.outputs.vulnerability-report }}'
 
+    - name: Process report
+      if: ${{ !cancelled() }}
```

Comment:
We might need to test if this also works for calling a reusable workflow in the `test-vulnerabilities` contains a failed job.